tests/test_client.py

```diff
@@ -598,3 +600,75 @@ def test_shared_blocks(client: Union[LocalClient, RESTClient], agent: AgentState
     # cleanup
     client.delete_agent(agent_state1.id)
     client.delete_agent(agent_state2.id)
+
+
+@pytest.fixture
+def cleanup_agents():
+    created_agents = []
+    yield created_agents
+    # Cleanup will run even if test fails
+    for agent_id in created_agents:
+        try:
+            client.delete_agent(agent_id)
+        except Exception as e:
+            print(f"Failed to delete agent {agent_id}: {e}")
+
+
+def test_initial_message_sequence(client: Union[LocalClient, RESTClient], agent: AgentState, cleanup_agents: List[str]):
```

Comment:
@sarahwooders tests should be self-explanatory for guidance on new kwarg `include_initial_boot_message`